### ISSUE TITLE
job-shell: no error on bad system current working directory

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#2679]
Per comments from @grondo in PR #2675 

```
$ flux mini run --setattr="system.cwd=/foo" pwd
/tmp

Should see an error about "unable to chrdir to /foo, going to /tmp instead".
```

It ends up that when the `job-exec` module runs the shell, it gets the error "Couldn't change dir to /foo: going to /tmp instead: No such file or directory", so the `job-shell` is run in `/tmp`.

The job shell mistakenly does not call `flux_cmd_setcwd()` to set the CWD, so it will never receive an error of `Couldn't change dir to ...`.  Since it's already in `/tmp`, `pwd` just outputs `/tmp` and it appears to `chdir()` without receiving the appropriate error message.

We should call `flux_cmd_setcwd()` in the job shell so we get the expected error message.

**Comment by grondo** [comment]
Since the job-shell runs as the user, it should be the process which calls `chdir()` so that it is running in the cwd of the job. Sorry, I should have realized this before making the above comment. Then each individual task doesn't need to separately `chdir()` (with the resulting duplicated error messages if there's an error doing so).

So the shell itself should try to `chdir()` to the working directory set in the jobspec, in case the `chdir()` set by the `job-exec` module failed, and it can then generate the expected error message using the shell log faciliity.

**Comment by chu11** [comment]
> So the shell itself should try to chdir() to the working directory set in the jobspec, in case the chdir() set by the job-exec module failed, and it can then generate the expected error message using the shell log faciliity.

Ahh, this makes sense.  I just pushed a fix into #2675, but I'll back it out.  I'll fix it in a different PR.
